### ISSUE TITLE
Vagrant 2.2.6 on Catalina fails to access (or ask to access) the Documents dir

### ISSUE THREAD

**Issue by no-simpler** [hashicorp/vagrant#11128]
macOS Catalina 10.15
Vagrant 2.2.6 (installed via Homebrew)

Whenever I do `vagrant up`, the process fails around this spot:

```
==> default: Destroying VM and associated drives...
/opt/vagrant/embedded/gems/2.2.6/gems/vagrant-2.2.6/plugins/providers/virtualbox/action/clean_machine_folder.rb:28:in `[]': Operation not permitted - /Users/username/Documents/Virtual Machines.localized (Errno::EPERM)
```

I understand, that Catalina introduced new protections for directories within user's home. I have granted the relevant permission to VirtualBox, but I see no way to manually add permission for Vagrant (see screenshot, plus button is grayed out).

<img width="344" alt="Screen Shot 2019-10-16 at 7 23 52 AM" src="https://user-images.githubusercontent.com/30579893/66888139-111adb00-efe6-11e9-9c1f-820473991f75.png">

Am I missing something here, or is this being worked on currently?

**Comment by briancain** [comment]
Hey there @no-simpler - thanks for reporting the issue. Unfortunately I don't think there's much we can do as Vagrant to give VirtualBox permission without users manually going into their settings to give VirtualBox access. I think we can print a more user friendly error message however, and put some help in our documentation about what might be required with Catalina and its security settings.

**Comment by no-simpler** [comment]
Sure thing. For anyone wondering, the simplest workaround is to move virtual machine directory to a non-protected location in the VirtualBox preferences. For me, `~/.vms` worked just fine.

**Comment by jbonhag** [comment]
Hi there @no-simpler -- I was investigating this issue a bit and I wanted to follow up.  I've recreated a similar environment as yours on macOS Catalina 10.15.1 (19B88):

* Vagrant 2.2.6 installed via Homebrew
* VirtualBox default machine folder set to `/Users/username/Documents/Virtual Machines.localized`

The first time I ran `vagrant up`, macOS prompted me to allow Terminal to access the Documents folder, which I granted, and after that I didn't have any issues with Vagrant permissions.

From your screenshot, I'm assuming that you're running Vagrant through the Hyper terminal emulator.  Would you be willing to enable Documents access for the Hyper app? That might eliminate the need to do more fine-grained permissions tuning.

These are the only app-specific permissions I have currently:

<img width="365" alt="Screenshot 2019-12-03 16 27 01" src="https://user-images.githubusercontent.com/6937257/70091253-bf57ff80-15e9-11ea-8103-14b76c8fbfec.png">

Looking forward to hearing back from you.  In the meantime I'll dig deeper into the more specific permissions.  Cheers!

**Comment by no-simpler** [comment]
Hey, @jbonhag, I went ahead and tried to retrace my steps on this one.

First of all, when I opened Virtualbox preferences and selected `<reset to default>` in the 'Default Machine Folder' field, it auto-populated to `/Users/username/VirtualBox VMs`. If memory serves, the default location used to be something like `~/Documents/Virtual Machines`. Perhaps the Virtualbox developers have changed this default in response to Catalina's new restrictions.

Otherwise, you are right: Hyper is my terminal, and whenever the Virtualbox VMs are under the protected `Documents` directory, the original error repeats if and only if the terminal itself does not have permissions for said directory.

It's funny how the culprit was actually in the dead center of my own original screenshot...
